portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -44,23 +89,26 @@ public void doGet(HttpServletRequest request, HttpServletResponse response) thro
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
     PreparedQuery results = datastore.prepare(query);
 
-    // Check all entities returned and store the relevant ones in comments
     comments = new ArrayList<>();
 
+    // Check all entities returned and store the relevant ones in comments
     for (Entity entity : results.asIterable()) {
 
-      // All fields of entity
-      // TODO: Use the timestamp field and show the time the comment was posted
-      long id = entity.getKey().getId();
-      String text = (String) entity.getProperty("text");
-      long timestamp = (long) entity.getProperty("timestamp");
+      // Skip all comments that belong to other pages
       String commentPage = (String) entity.getProperty("page");
-
-      // Only show comments on that recipe
       String currentPage = request.getHeader("referer").replace("\n", "");
-      if (commentPage.equals(currentPage)){
-        comments.add(text);
+      if (!commentPage.equals(currentPage)){
+        continue;
       }
+
+      Comment newComment = new Comment();
```

Comment:
the constructor should have parameters instead of the setters.

Reply:
Done!